### DISCUSSION TITLE
MasterNode installation location

### DISCUSSION THREAD

**Discussion by jaystile** [dragonflyoss/Dragonfly2#556] [Q&A]
If you have an existing Kubernetes cluster would you advise installing the MasterNode inside the cluster? Or would you recommend installing it externally to the cluster?

When I was examining the [dragonfly & k8s quick start guide](https://github.com/dragonflyoss/Dragonfly2/blob/main/docs/en/ecosystem/Kubernetes-with-Dragonfly.md) I was having trouble determining if this was only setting up peers within the k8s cluster or if it was setting up the entire ecosystem including the MasterNode.

**Comment by ish-xyz** [comment]
I'd keep the "control plane" separate. Manager, CDN, Mysql, etc.
Also, you could run it "headless" if you want. I don't think the UI/WebApp is very useful. At least not to me.
If we're talking about Dragonfly != v2 then you could keep the supernodes on the cluster, but it really depends on how your infrastructure looks like.

**Reply by jaystile** [reply]
Thank you for the information. It took me time to figure out that MasterNodes are a concept in version 1 and not version 2. It sounds like your recommendation is install the Manager, CDN, and MySQL externally from the Kubernetes cluster. I think that means only the dfdaemon would run inside the Kubernetes cluster. 

I also found the UI to not be useful, but the REST API was pretty good. The UI confused me a little bit as when you login without credentials it indicates you are an 'Admin', but you cannot execute any commands. I needed to logout and sign in as 'root'.

**Reply by ish-xyz** [reply]
I think it really depends on your infrastructure.
In our use case we're going to run only the CDN, the scheduler + obviously the dfdaemons. I'm gonna keep everything together and replicated for each cluster, even if we have quite a lot of them. 
That means that every cluster will have its own scheduler (sts), cdn (sts) and dfdaemons (daemonset).

In the future I might move the CDN in a dedicated cluster and by doing that I'll be able to have only "one" seeder for multiple clusters, this should allow us to save storage space and optimise the image distribution. This should work also for the scheduler, but it might be tricky to implement in our environments.


How many clusters are you going to have that uses Dragonfly?
